drgn_tools/eventpoll.py

```diff
@@ -0,0 +1,228 @@
+
+
+def for_each_epitem_in_monitor_list(eventpoll: Object) -> Iterator[Object]:
+    """
+    Iterate through epitem(s) in an eventpoll's monitor list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of ``struct epitem *``.
+    """
+    for epitem in rbtree_inorder_for_each_entry(
+        "struct epitem", eventpoll.rbr.rb_root.address_of_(), "rbn"
+    ):
+        yield epitem
+
+
+def for_each_epitem_in_ready_list(eventpoll: Object) -> Iterator[Object]:
+    """
+    Iterate through epitem(s) in an eventpoll's ready list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of ``struct epitem *``.
+
+    """
+
+    for epitem in list_for_each_entry(
+        "struct epitem", eventpoll.rdllist.address_of_(), "rdllink"
+    ):
+        yield epitem
+
+
+def for_each_epitem_in_overflow_list(eventpoll: Object) -> Iterator[Object]:
+    """
+    Iterate through epitem(s) in an eventpoll's overflow list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of ``struct epitem *``.
+
+    """
+    epitem = eventpoll.ovflist
+    while epitem.value_() != 0xFFFFFFFFFFFFFFFF:
+        yield epitem
+        epitem = epitem.next
+
+
+def for_each_file_in_monitor_list(
+    eventpoll: Object,
+) -> Iterator[Tuple[int, Object]]:
+    """
+    Iterate through files in an eventpoll's monitor list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of file descriptor and ``struct file *``.
+    """
+    for epitem in for_each_epitem_in_monitor_list(eventpoll):
+        yield epitem.ffd.fd.value_(), epitem.ffd.file
+
+
+def for_each_file_in_ready_list(
+    eventpoll: Object,
+) -> Iterator[Tuple[int, Object]]:
+    """
+    Iterate through files in an eventpoll's ready list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of file descriptor and ``struct file *``.
+    """
+    for epitem in for_each_epitem_in_ready_list(eventpoll):
+        yield epitem.ffd.fd.value_(), epitem.ffd.file
+
+
+def for_each_file_in_overflow_list(
+    eventpoll: Object,
+) -> Iterator[Tuple[int, Object]]:
+    """
+    Iterate through files in an eventpoll's overflow list.
+
+    :param eventpoll: ``struct eventpoll *``
+    :returns: Iterator of file descriptor and ``struct file *``.
+    """
+    for epitem in for_each_epitem_in_overflow_list(eventpoll):
+        yield epitem.ffd.fd.value_(), epitem.ffd.file
+
+
+def print_file_list(file_list: List[Tuple[int, Object]]) -> None:
+    """
+    Print information about files in a given list.
+
+    param file_list: list of file descriptor(s) and ``struct file *``
+    """
+    tbl = FixedTable(
+        [
+            "fd:>",
+            "(struct file *):016x",
+            "(struct dentry *):016x",
+            "(struct inode *):016x",
+            "path:>",
+        ]
+    )
+    for fd, file in file_list:
+        path = d_path(file.f_path)
+        escaped_path = escape_ascii_string(path, escape_backslash=True)
```

Comment:
I suppose the watched file should be socket or pipe? Wondering what kind of output will it be? Can you show an example of the output?